libcore/FileUtils.vala

```diff
@@ -552,6 +552,15 @@ namespace PF.FileUtils {
                 return false;
         }
     }
+
+    public bool location_is_in_trash (GLib.File location) {
+        var uri = location.get_uri ();
+
+        return (uri.has_prefix ("trash") ||
```

Comment:
Shouldn't that test if the file has an URI scheme, like the C code did?

Reply:
I guess so - this is just a bit quicker.